src/picture.rs

```diff
@@ -103,15 +102,30 @@ pub unsafe extern "C" fn dav1d_default_picture_alloc(
     let y_sz = (y_stride * aligned_h as isize) as usize;
     let uv_sz = (uv_stride * (aligned_h >> ss_ver) as isize) as usize;
     let pic_size = y_sz + 2 * uv_sz;
-    let buf = rav1d_mem_pool_pop(
-        cookie as *mut Rav1dMemPool,
-        pic_size + RAV1D_PICTURE_ALIGNMENT - ::core::mem::size_of::<Rav1dMemPoolBuffer>(),
-    );
-    if buf.is_null() {
-        return Rav1dResult::<()>::Err(ENOMEM).into();
+
+    let pool = &*cookie.cast::<Arc<MemPool<u8>>>();
```

Comment:
What do you mean by load-bearing clone?  I'd rather do a safe `Arc::clone` instead of an `unsafe` `Arc::increment_strong_count`.

Reply:
It's not clear that the pointer is a borrowed Arc, or even which copy of the Arc that borrow is of. I'd prefer that pointer is an owned copy of the Arc that you then manage access to. Just because `increment_strong_count` is unsafe doesn't mean it's not the right choice here. The unsafe deref and reference of the pointer is hiding a lot of pitfalls.